src/App.js

```diff
@@ -1,64 +1,77 @@
 import React, { Component } from "react";
 import "./App.css";
 
-import Card from "./components/Card/Card";
 import AddCard from "./components/AddCard/AddCard";
+import AllCards from "./components/AllCards/AllCards";
+import Search from "./components/Search/Search";
 
 class App extends Component {
-  state = {
-    cards: [
-      {
-        id: 1,
-        name: "John",
-        email: "john@gmail.com"
-      },
-      {
-        id: 2,
-        name: "Rob",
-        email: "rob@gmail.com"
-      }
-    ],
-    showAddCard: false,
-    lastId: 2
-  };
+  constructor() {
+    super();
+    this.state = {
+      cards: [],
+      showAddCard: false,
+      lastId: 2,
+      filterText: ""
+    };
+  }
+
+  componentWillMount() {
+    this.setState({
+      cards: [
+        {
+          id: 1,
+          name: "John",
+          email: "john@gmail.com"
+        },
+        {
+          id: 2,
+          name: "Rob",
+          email: "rob@gmail.com"
+        }
+      ]
+    });
+  }
 
   onShowAddCard = () => {
     const { showAddCard } = this.state;
     this.setState({ showAddCard: !showAddCard });
   };
 
-  onAddCard = (card) => {
+  onAddCard = card => {
```

Comment:
Can you describe what's the difference between setting this method as fat arrow and regular function `onAddCard(card) {` ?